dev/integration_tests/ios_platform_view_tests/lib/main.dart

```diff
@@ -154,3 +166,65 @@ class _FocusTestPageState extends State<FocusTestPage> {
     );
   }
 }
+
+/// A page to test a platform view in an alert dialog prompt is still tappable.
+/// See [this issue](https://github.com/flutter/flutter/issues/118366).
+class ZOrderTestPage extends StatefulWidget {
+  const ZOrderTestPage({super.key});
+
+  @override
+  State<ZOrderTestPage> createState() => _ZOrderTestPageState();
+}
+
+class _ZOrderTestPageState extends State<ZOrderTestPage> {
+  bool _showBackground = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Platform view z order test'),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Visibility(
+                visible: _showBackground,
+                child: const SizedBox(
+                  width: 500,
+                  height: 500,
+                  child: UiKitView(
```

Comment:
You can add a semantics node when "showBackground is true", then in the XCTest, you can search for that node instead of delaying.

Reply:
The issue I had is that for some reason it fails to query the second platform view which is controlled by `showBackground` (Refer to https://github.com/flutter/flutter/pull/120844/files#diff-3e48a5714c1ea61112bebbd881876bea61cf86f065890eda9b019360d62cee97R85-R88)

Are you saying I should add a "semantics node" for the second platform view to work?

Reply:
Offline discussion, it is not possible as dialog seems to remove other semantics nodes from the tree. I couldn't think of another way better than using the delay in the test.